src/ufixed.jl

```diff
@@ -47,6 +47,9 @@ convert{U<:UFixed}(::Type{U}, x::Real) = _convert(U, rawtype(U), x)
 _convert{U<:UFixed,T}(::Type{U}, ::Type{T}, x)       = U(round(T, widen1(rawone(U))*x), 0)
 _convert{U<:UFixed  }(::Type{U}, ::Type{UInt128}, x) = U(round(UInt128, rawone(U)*x), 0)
 
+rem{T<:UFixed}(x::T, ::Type{T}) = x
+rem{T<:UFixed}(x::UFixed, ::Type{T}) = reinterpret(T, rem(Integer(round((rawone(T)/rawone(x))*reinterpret(x))), rawtype(T)))
```

Comment:
Very interesting. I looked at three "inner" implementations:
```jl
@inline iscale1{T}(::Type{T}, x) = round(UInt, (rawone(T)/rawone(x))*reinterpret(x))
@inline iscale1a{T}(::Type{T}, x) = unsafe_trunc(UInt, round((rawone(T)/rawone(x))*reinterpret(x)))
@inline function iscale2{T}(::Type{T}, x)
    d, r = divrem(widemul(rawone(T), reinterpret(x)), rawone(x))
    d + oftype(d, r >= rawone(x)>>1)
end
```
The fastest seems to be `iscale1a`. Here's its LLVM:
```jl
julia> @code_llvm iscale1a(UFixed12, one(UFixed10))

define i64 @julia_iscale1a_70902(%jl_value_t*, %UFixed.65*) #0 {
top:
  %2 = getelementptr inbounds %UFixed.65, %UFixed.65* %1, i64 0, i32 0
  %3 = load i16, i16* %2, align 2
  %4 = uitofp i16 %3 to double
  %5 = fmul double %4, 0x40100300C0300C03
  %6 = call double @llvm.rint.f64(double %5)
  %7 = fptoui double %6 to i64
  ret i64 %7
}
```
`iscale1` looks similar except the last expression is a `call`, and that call turns out to involve a branch.